Create CSS Entry for clamp() (#574)
Codecademy/docs: content/css/concepts/math-functions/terms/clamp/clamp.md

---
Title: 'clamp()'
Description: 'Restricts a given value between an upper and lower bound.'
Subjects:
  - 'Web Development'
  - 'Web Design'
Tags:
  - 'Responsive'
  - 'Functions'
CatalogContent:
  - 'learn-css'
  - 'paths/front-end-engineer-career-path'
  - 'paths/full-stack-engineer-career-path'
---

The `clamp()` function restricts a given value between an upper and lower bound. In this way, it acts like a combination of the [`min()`](https://www.codecademy.com/resources/docs/css/math-functions/min) and [`max()`](https://www.codecademy.com/resources/docs/css/math-functions/max) functions.

## Syntax

```pseudo
<property>: clamp(<lower bound>, <value>, <upper bound>)
```

> The value will be the given `<value>` as long as it is between the `<lower bound>` and `<upper bound>`. The minimum value returned will be the `<lower bound>` and the maximum value returned will be the `<upper bound>`.

Reviewer: ```suggestion
The `<value>` is valid as long as it is between the `<lower bound>` and `<upper bound>`. This ensures that the `<property>` has a minimum and maximum `<value>`.
```

> Each argument of the `clamp()` function can be one of the following:

Reviewer: ```suggestion
Each argument to `clamp()` can be a combination of [CSS units](https://www.codecademy.com/resources/docs/css/units), or evaluations of them, using the following:
```

> - Length values: `10px`

Reviewer: ```suggestion
- Absolute units: `10px`
```

> - Relative length values: `40vh`

Reviewer: ```suggestion
- Relative units: `40vh`
```

> - Nested math function values: `min(20vw, 200px)`
> - Math expressions using arithmetic operators: `40vw - 20px`

Reviewer: ```suggestion
- Math expressions: `40vw - 20px`
- Nested math function values: `min(20vw, 200px)`
```

## Example

```css
.display-font {
  font-size: clamp(2rem, 3vw, 5rem);
}
```

> This results in a font size for `.display-font` elements that will change with the viewport width, but not smaller than 2rem or larger than 5rem.

Reviewer 2: ```suggestion
This results in a font size for `.display-font` elements that will span `3%` of the viewport's width, not smaller than `2rem`, or larger than `5rem`.
```

> This results in a font size for `.display-font` elements that will span `3%` of the viewport's width, not smaller than `2rem`, or larger than `5rem`.

Reviewer: ```suggestion
In the example above, `.display-font` elements will have their `font-size` clamped down to span `3%` of the viewport's width. However, it can be no more than `5rem` and no less than `2rem`.
```
